### PR TITLE
Feature/add-exercises

### PR DESCRIPTION
**Pull request by thegunner686**
# Notes
- Created ExerciseBlock
- Created button for adding exercises to a workout
- Navigation to EditExercisePage
- Loads in the currently selected workout data and creates a real-time listener for changes
- Everything persisted in Supabase

# Testing
- Tested on Expo on iPhone

### CODE REVIEW

**Review by pajsimbulan — COMMENTED** [review]
these are cleannn

File: src/components/ExerciseSearchBar.tsx [hunk 1]
```diff
@@ -64,7 +64,7 @@ const ExerciseSearchBar: React.FC<ExerciseSearchBarProps> = ({ onSelectExercise
         />
       </View>
       <View className="divide-y-10 invisible  h-60 w-11/12 overflow-scroll border border-slate-200 bg-slate-50">
-        <ScrollView>
+        <ScrollView keyboardShouldPersistTaps="always">
```
Comment (pajsimbulan) [inline src/components/ExerciseSearchBar.tsx:67]: forgot the keyboard thing, thanks for adding it for me

File: src/screens/EditWorkout/EditWorkoutPage.tsx [hunk 2]
```diff
@@ -1,21 +1,86 @@
...
+import { Exercise, PLACEHOLDER_EXERCISE_NAME } from '../../types/Exercise';
+import ExerciseBlock from '../../components/ExerciseBlock';
+import { updateExerciseInWorkout, useEditableWorkout } from '../../hooks/useEditableWorkout';
+import useSelectedWorkout from '../../hooks/useSelectedWorkout';
 
 type EditWorkoutPageProps = NativeStackScreenProps<NavigatorParamList, 'EditWorkoutPage'>;
 
 const EditWorkoutPage: React.FC<EditWorkoutPageProps> = ({ navigation }) => {
-  const navigateToEditExercisePage = () => {
+  const [selectedWorkout, setSelectedExercise] = useSelectedWorkout((state) => [
```
Comment (pajsimbulan) [inline src/screens/EditWorkout/EditWorkoutPage.tsx:14]: const [selectedWorkout, **setSelectedWorkout**] ?

**Review by pajsimbulan — APPROVED** [review]
dis sum juicy pr.  LJTM (looks juicy to me).

File: src/screens/EditWorkout/EditWorkoutPage.tsx (hunk 2 above)
Comment (thegunner686) [reply src/screens/EditWorkout/EditWorkoutPage.tsx:14]: `selectedWorkout` is to display the workout, `setSelectedExercise` is for setting the exercise to view when we navigate to the edit exercise page.

**Pull request merged**